app/src/main/java/com/kamron/pogoiv/MainActivity.java

```diff
@@ -295,13 +295,42 @@ private void startPokeFly() {
         Intent intent = Pokefly.createIntent(this, trainerLevel, statusBarHeight, batterySaver);
         startService(intent);
 
-        pokeFlyRunning = true;
-
         if (settings.shouldLaunchPokemonGo()) {
             openPokemonGoApp();
         }
     }
 
+    private boolean isPokeFlyRunning() {
+        ActivityManager manager = (ActivityManager) getSystemService(Context.ACTIVITY_SERVICE);
+        for (ActivityManager.RunningServiceInfo serviceInfo : manager.getRunningServices(Integer.MAX_VALUE)) {
+            if (Pokefly.class.getName().equals(serviceInfo.service.getClassName())) {
+                return true;
+            }
+        }
```

Comment:
I don't like that code too, but it's one of the most reasonable ways to accomplish that check.
It doesn't seem to be computationally intensive but I'm open to ideas and suggestions on this!